mbf_abstract_nav/include/mbf_abstract_nav/abstract_action_base.hpp

```diff
@@ -175,11 +179,12 @@ class AbstractActionBase
   {
     uint8_t slot = goal_handle.getGoal()->concurrency_slot;
 
-    boost::lock_guard<boost::mutex> guard(slot_map_mtx_);
+    slot_map_mtx_.lock();
     typename ConcurrencyMap::iterator slot_it = concurrency_slots_.find(slot);
-    if (slot_it != concurrency_slots_.end())
+    slot_map_mtx_.unlock();
+    if (slot_it != concurrency_slots_.end() && slot_it->second.execution)
     {
-      concurrency_slots_[slot].execution->cancel();
+      slot_it->second.execution->cancel();
```

Comment:
while we have ruled out that this will be a nullptr, it seems to me that we still might face a race condition  the cancel might be called before the start - missing the cancel request

Reply:
We need to define what's the expected results if a new goal comes right after the cancel. In the particular case you mention, I think the new goal superseeds the cancel and we proceed to execute it